lang/java/avro/src/test/java/org/apache/avro/TestAvro4078.java

```diff
@@ -0,0 +1,21 @@
+package org.apache.avro;
+
+import org.apache.avro.specific.SpecificData;
+import org.junit.Test;
+
+/**
+ * Unit test for demonstrating specific data potential issue.
+ */
+public class TestAvro4078 {
+
+  public static final Schema FULLNAME_SCHEMA = (new Schema.Parser()).parse("{\n" + "     \"type\": \"record\",\n"
+      + "     \"namespace\": \"org.apache.avro\",\n" + "     \"name\": \"FullName\",\n" + "     \"fields\": [\n"
+      + "       { \"name\": \"first\", \"type\": \"string\" },\n"
+      + "       { \"name\": \"last\", \"type\": \"string\" }\n" + "     ]\n" + "}");
+
+  @Test
+  public void testClassLoad() {
+    System.err.println(FULLNAME_SCHEMA);
+    System.err.println(SpecificData.get().getClass(FULLNAME_SCHEMA));
```

Comment:
can you please add `assertNotNull(SpecificData.get().getClass(FULLNAME_SCHEMA))` to make sure that the test only passes when its successfully able to load the class

Reply:
Done!
This shows the problem in the test case too!
The static initialization of FullName `READER$` and `WRITER$` fields lead to a hang!